doc/make.py

```diff
@@ -159,10 +159,10 @@ def _get_page_title(self, page):
         Open the rst file `page` and extract its title.
         """
         fname = os.path.join(SOURCE_PATH, f"{page}.rst")
-        option_parser = docutils.frontend.OptionParser(
-            components=(docutils.parsers.rst.Parser,)
+        doc = docutils.utils.new_document(
```

Comment:
Can you use something like this instead? https://stackoverflow.com/a/75996218

Reply:
We could, but my understanding from [the docs](https://github.com/docutils/docutils/commit/6548b56d9ea9a3e101cd62cfcd727b6e9e8b7ab6#diff-a033583f6ace19fed2adc108b1c130e17ea00e1afee38c22c993cb477ad27a5fR453) is that this version allows customization through a configuration file, whereas the SO answer does not. So I'd lean on keeping it like it is, but I'm happy to change the PR if you'd prefer.

(This is assuming you're not talking about supporting docutils prior to 0.18. Otherwise let me know.)

Reply:
Yeah I was referring to using `get_default_settings`. I would prefer using more "public" APIs from docutils rather then digging into `docutils.core`